### DISCUSSION TITLE
Files available publicly if using S3

### DISCUSSION THREAD

**Discussion by jperestrelo** [docmost/docmost#561] [Help / Questions] [answered]
First off all, thanks for this project, it really looks like what we are looking for.

BUG ?
When uploading a file using S3 the file becomes publicly available.
I noticed this previous issue https://github.com/docmost/docmost/issues/35 but its not working as mentioned.

On digitalocean space file it shows private and when I try to get the file using the endpoint its not accessible but when I use the direct link from my hosted docmost apps the file its public.

This is very sensitive, since we cant share some of company files publicly.

Thanks in advance.

**Comment by Philipinho** [comment]
Page attachments should be private. Avatars can be accessed if the link is known.
Please confirm which.

**Comment by jperestrelo** [comment]
File attachments to page, they can be access by anyone from a url like this:
https://mydocmostsite.com/api/files/0193c60f-50a3-704f-a999-29b5a09a8d37/some-file.pdf

**Comment by Philipinho** [comment]
I am still unable to reproduce.

i. Please state the version of Docmost you are running.
ii. Confirm that you have tried from a separate browser without any valid Docmost sessions.

**Comment by jperestrelo** [comment]
Im using latest version: 0.6.1
I tried in incognito and I can reproduce it

Im using digitalocean spaces bucket with this config variables:
STORAGE_DRIVER=s3
AWS_S3_ACCESS_KEY_ID=MYKEY
AWS_S3_SECRET_ACCESS_KEY=MYSECRET
AWS_S3_REGION=ams3
AWS_S3_BUCKET=MYBUCKET
AWS_S3_ENDPOINT=https://ams3.digitaloceanspaces.com

**Reply by Philipinho** [reply]
If your server is accessible over the internet, please forward an example attachment link to my email (in bio).

**Reply by jperestrelo** [reply]
done, sent.
btw, when I was checking before sending to you it showed forbidden, but then I refreshed a few times and it was public again.
Now I keep trying to reproduce the forbidden and its not showing as such

**Reply by Philipinho** [reply]
I received the link. I will let you know my findings.

**Reply by Philipinho** [reply]
I could access the link you mailed. I realized the attachment was cached by your CDN. It was subsequently served from the CDN after the first access.

I pushed a fix to prevent such behaviour: https://github.com/docmost/docmost/pull/562.
It should be available in v0.6.2.

**Reply by Philipinho** [reply, marked as answer]
You can now upgrade to [v0.6.2](https://github.com/docmost/docmost/releases/tag/v0.6.2).
I look forward to your feedback.

**Reply by jperestrelo** [reply]
Perfect! I just upgraded, and now I can see:
`{"message":"Unauthorized","statusCode":401}`
Thanks a lot @Philipinho. 

Since Docmost is an early project I was a bit hesitant to use it for our company documentation but our interaction really gave me the confidence to move forward with it. We’ll start migrating all our business processes and knowledge base to here.

I wish I had time to contribute with code, but at least I will try to give feedback or report bugs if I find them.

Thanks again for this project—it’s going to be incredibly useful for us.
Cheers!
